File: unit-4/sprint-3/day-1/you/app/src/main/java/com/example/paisamanage/remote/Network.java
package com.example.paisamanage.remote;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Network {

    private static final String BASE_URL = "http://13.232.169.202:8080/users/";

    public static Retrofit getRetrofit(){
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(new OkHttpClient.Builder().build())
                .build();
        return retrofit;
    }

}

File: unit-4/sprint-4/day-4/assignments/app/src/main/java/com/example/moviestask/remote/network.kt
package com.example.moviestask.remote

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Singleton

@InstallIn(SingletonComponent::class)
@Module
object network {

    val httploggingintercepter= HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY)

    @Singleton
    @Provides
     fun getretrofit(): Retrofit {
        return Retrofit.Builder()
            .baseUrl( "https://gist.githubusercontent.com/").addConverterFactory(GsonConverterFactory.create())
            .client(OkHttpClient.Builder().addInterceptor(httploggingintercepter).build()).build()
    }

    @Singleton
    @Provides
    fun getapiservice(retrofit: Retrofit) :ApiService = retrofit.create(ApiService::class.java)


}